psi4/driver/task_planner.py

```diff
@@ -0,0 +1,257 @@
+#
+# @BEGIN LICENSE
+#
+# Psi4: an open-source quantum chemistry software package
+#
+# Copyright (c) 2007-2022 The Psi4 Developers.
+#
+# The copyrights for code used from other parties are included in
+# the corresponding files.
+#
+# This file is part of Psi4.
+#
+# Psi4 is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, version 3.
+#
+# Psi4 is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License along
+# with Psi4; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# @END LICENSE
+#
+
+__all__ = ["task_planner"]
+
+import os
+import copy
+import logging
+from typing import Dict, Tuple, Union
+
+from qcelemental.models import DriverEnum
+from psi4.driver import p4util, pp
+from psi4.driver.task_base import AtomicComputer
+from psi4.driver.driver_findif import FiniteDifferenceComputer
+from psi4.driver.driver_nbody import ManyBodyComputer
+from psi4.driver.driver_cbs import CompositeComputer, composite_procedures, cbs_text_parser
+from psi4.driver.driver_util import negotiate_derivative_type, negotiate_convergence_criterion
+
+logger = logging.getLogger(__name__)
+
+TaskComputers = Union[AtomicComputer, CompositeComputer, FiniteDifferenceComputer, ManyBodyComputer]
+
+
+def _expand_cbs_methods(method: str, basis: str, driver: DriverEnum, **kwargs) -> Tuple[str, str, Dict]:
```

Comment:
docstring to say what the function does, please. I can guess based on the name.

Reply:
Added 

```def _expand_cbs_methods(method: str, basis: str, driver: DriverEnum, **kwargs) -> Tuple[str, str, Dict]:
    """Sort out the user input method string into recognized fields. Handles cases like:

    * (i) `"mp2"` --> passes through;
    * (ii) `"mp2/cc-pvdz"` --> broken into method and basis fields;
    * (iii) `"mp2/cc-pv[d,t]z"` --> processed into method="cbs" and CBSMetadata spec;
    * (iv) `method="cbs", cbsmeta=CBSMetadata` --> passes through.

    """
```